core/matrix/csr.cpp

```diff
@@ -672,6 +675,23 @@ void Csr<ValueType, IndexType>::inv_scale_impl(const LinOp* alpha)
 }
 
 
+template <typename ValueType, typename IndexType>
+void Csr<ValueType, IndexType>::add_scaled_identity_impl(const LinOp* const a,
+                                                         const LinOp* const b)
+{
+    bool has_diags{false};
+    this->get_executor()->run(
+        csr::make_check_diagonal_entries(this, has_diags));
```

Comment:
should it always check?

Reply:
Right now, we don't have a mechanism for the user to tell us whether it's needed. If we add a property, I'm not sure where it should go right now. So for now, I'm thinking of always carrying out the check, until we have a good general way to handle matrix properties or pre-condition checks. Or until someone says it's too slow with the check.